sched/demo/main.go

```diff
@@ -30,21 +33,50 @@ func main() {
 	}()
 
 	go func() {
-		scheduleWork(workCh, cluster, distributor)
+		scheduleWork(workCh, cluster, distributor, saga)
 		wg.Done()
 	}()
 
 	wg.Wait()
+
+	ids, err := saga.Startup()
+
+	// we are using an in memory saga here if we can't get the active sagas something is
+	// very wrong just exit the program.
+	if err != nil {
+		fmt.Println("ERROR getting active sagas ", err)
+		os.Exit(2)
+	}
+
+	completedSagas := 0
+
+	for _, sagaId := range ids {
+
+		sagaState, err := saga.RecoverSagaState(sagaId, s.ForwardRecovery)
+		if err != nil {
+			// For now just print error in actual scheduler we'd want to retry multiple times,
+			// before putting it on a deadletter queue
+			fmt.Println(fmt.Sprintf("ERROR recovering saga state for %s: %s", sagaId, err))
+		}
+
+		// all Sagas are expected to be completed
+		if !sagaState.IsSagaCompleted() {
```

Comment:
I plan to change it once we build the Node component.  The InMemoryNode will randomly generate errors so we can verify the system works correctly with dropped messages or downed nodes etc...